### PR TITLE
Fix multiple preconfirmations for the same slot in demo

### LINKED ISSUE #107: Handle two or more pre-confirmations at the same slot in the demo
There is an edge case in the frontend demo where sending two pre-confirmations rapidly in the same slot can cause the second one to fail with `NonceTooLow` error.
This is due to fact the nonce is obtained from the provider i.e. the execution state, which hasn't processed the pre-confirmed transaction yet, and as such the nonce of same transaction is the same of the first one.

In order to solve this, in the frontend we should increment the nonce of the account upon receiving the commitment from the sidecar

### PR DESCRIPTION
**Pull request by thedevbirb**
Closes #107. Note that this doesn't really allow spamming continuously the button since there is a little delay between update of React state variables.

## Summary by CodeRabbit

- **New Features**
	- Enhanced the `Home` component for better handling of preconfirmation requests and wallet integration.
	- Introduced constants for secure interactions with the Kurtosis development network.

- **Improvements**
	- Updated the transaction handling to ensure accuracy and responsiveness to blockchain events.
	- Streamlined payload creation for preconfirmation requests and improved function modularity.

- **Bug Fixes**
	- Improved event handling logic to filter out outdated requests.

- **Chores**
	- Reformatted the startup script for better readability and added a delay before opening the browser to enhance user experience.

### CODE REVIEW

File: bolt-web-demo/frontend/src/lib/constants.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Test private key, for which address[0] holds 1000 ETH in the Kurtosis devnet
+export const PRIVATE_KEY =
+  "39725efee3fb28614de3bacaffe4cc4bd8c436257e2c8bb887c4b5c4be45e76d";
```
[the remark this answers is not in the document]
Comment (thedevbirb) [reply bolt-web-demo/frontend/src/lib/constants.ts:3]: yolo

File: scripts/start-demo.sh [hunk 2]
```diff
@@ -25,10 +25,12 @@ trap 'quit_all' SIGINT
 # Start the commands in the background
 for command in "${commands[@]}"; do
   echo "Starting: $command"
-  eval "$command" &  # Use eval to handle complex commands with CD and chaining
+  eval "$command" & # Use eval to handle complex commands with CD and chaining
 done
 
-# Open the browser 
+sleep 1
```
Comment (merklefruit) [inline scripts/start-demo.sh:31]: very nice DX improvement 😄

**Review by merklefruit — APPROVED** [review]
lgtm, can I try it on the demo on this branch?

**Comment by thedevbirb** [comment]
> lgtm, can I try it on the demo on this branch?



https://github.com/user-attachments/assets/0fcfbd78-1fc7-42a4-b5d2-39fba785fec0

**Comment by merklefruit** [comment]
Very nice :)

**Pull request merged**